torch/jit/__init__.py

```diff
@@ -1564,10 +1564,7 @@ def annotate(the_type, the_value):
     return the_value
 
 
-class Attribute(object):
-    def __init__(self, value, the_type):
-        self.value = value
-        self.type = the_type
+Attribute = collections.namedtuple('Attribute', ['value', 'type'])
```

Comment:
Maybe use __slots__ here instead of named tuple so that it can't now be indexed ?

Reply:
So, my understanding is that you should namedtuple if it's immutable, and slots if it's mutable. This class (and the one above) looked immutable, so it seemed like namedtuple was appropriate. What do you mean by "can't be indexed"?

Reply:
sorry by indexed i meant it can't be used like a tuple - which includes tuple

Reply:
I acknowledge your concern. I guess it depends on how much we care about enforcing immutability versus preventing indexing. I'm going to come down in favor of immutability.